apps/gdalalg_abstract_pipeline.cpp

```diff
@@ -0,0 +1,1973 @@
+/******************************************************************************
+ *
+ * Project:  GDAL
+ * Purpose:  gdal "raster/vector pipeline" subcommand
+ * Author:   Even Rouault <even dot rouault at spatialys.com>
+ *
+ ******************************************************************************
+ * Copyright (c) 2024-2025, Even Rouault <even dot rouault at spatialys.com>
+ *
+ * SPDX-License-Identifier: MIT
+ ****************************************************************************/
+
+#include "cpl_conv.h"
+#include "cpl_error_internal.h"
+#include "cpl_json.h"
+
+#include "gdalalg_abstract_pipeline.h"
+#include "gdalalg_raster_read.h"
+#include "gdalalg_raster_write.h"
+#include "gdalalg_tee.h"
+
+#include <algorithm>
+#include <cassert>
+
+//! @cond Doxygen_Suppress
+
+/* clang-format off */
+constexpr const char *const apszReadParametersPrefixOmitted[] = {
```

Comment:
I'm not reviewing this file because I assume it is all code moved from the raster and vector pipelines.
@rouault  Please let me know if you want me to go through it.

Reply:
yes the code was just moved from the .h to the .cpp file